core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalSumTests.java

```diff
@@ -46,8 +49,27 @@ protected InternalSum createTestInstance(String name, List<PipelineAggregator> p
 
     @Override
     protected void assertReduced(InternalSum reduced, List<InternalSum> inputs) {
-        double expectedSum = inputs.stream().mapToDouble(InternalSum::getValue).sum();
-        assertEquals(expectedSum, reduced.getValue(), 0.0001d);
+        double expectedSum = 0;
+        double compensation = 0;
+        for (InternalSum aggregation : inputs) {
+            double corrected = aggregation.value() - compensation;
+            double newSum = expectedSum + corrected;
+            compensation = (newSum - expectedSum) - corrected;
+            expectedSum = newSum;
+        }
+        assertEquals(expectedSum, reduced.getValue(), 0.000d);
+    }
+
```

Comment:
Why I have to modify this test case is because if there is a `Double.NEGATIVE_INFINITY` in the values to sum, `Kahan summation` with get `Double.NaN`, and naive summation will get `Double.NEGATIVE_INFINITY`, I'm not sure if this use case should be supported.

Reply:
We started rejecting this on new indices recently, but old indices might still have infinities. I think we should have special handling for these values in the aggregators?

Reply:
I pushed 15cdec228de94240e0a78b9dd2eed0b7f1f1aaac